From d1fc2a53e4e356da75e68325704d97d9af12f075 Mon Sep 17 00:00:00 2001
From: David Vrabel <david.vrabel@citrix.com>
Date: Wed, 24 Jul 2013 10:40:33 +0100
Subject: [PATCH 1/3] trace: include timestamp in trace records added by HVMOP_xentrace

TRC_GUEST trace records added by the HVMOP_xentrace op are not
timestamped.  Timestamping these records is useful if only TRC_GUEST
records are enabled.

Signed-off-by: David Vrabel <david.vrabel@citrix.com>
diff --git a/xen/arch/x86/hvm/hvm.c b/xen/arch/x86/hvm/hvm.c
index f0e1edc..5a4e080 100644
--- a/xen/arch/x86/hvm/hvm.c
+++ b/xen/arch/x86/hvm/hvm.c
@@ -6053,8 +6053,7 @@ long do_hvm_op(unsigned long op, XEN_GUEST_HANDLE_PARAM(void) arg)
              || (tr.event & ~((1u<<TRC_SUBCLS_SHIFT)-1)) )
             return -EINVAL;
 
-        /* Cycles will be taken at the vmexit and vmenter */
-        trace_var(tr.event | TRC_GUEST, 0 /*!cycles*/,
+        trace_var(tr.event | TRC_GUEST, 1 /*cycles*/,
                   tr.extra_bytes, tr.extra);
         break;
     }
